bin/clinvar_jsons/clinvar_jsons_shared_lib.py

```diff
@@ -0,0 +1,89 @@
+import gzip
+import json
+from collections import namedtuple
+
+
+class Trait:
+    def __init__(self, name):
+        self.name = name
+        self.xref_set = set()
+        self.count = 1
+
+    # def __eq__(self, other):
+    #     if isinstance(other, Trait):
+    #         return self.name == other.name and set(self.xref_set) == set(other.xref_set)
+    #     else:
+    #         return False
+    #
+    # def __hash__(self):
+    #     return hash((self.name, self.xref_set))
+    #
+    # def __ne__(self, other):
+    #     return not (self == other)
+
+    @property
+    def xref_string(self):
+        return "|".join(["{}/{}".format(xref.db, xref.id_)
+                         for xref in self.xref_set if xref.status.lower() == "current"])
+
+    def __str__(self):
+        return "{}\t{}\t{}".format(self.name, self.xref_string, self.count)
+
+
+TraitXref = namedtuple("TraitXref", ["db", "id_", "status"])
+
+
+def clinvar_jsons(filepath):
+    with gzip.open(filepath, "rt") as f:
+        for line in f:
+            line = line.rstrip()
+            yield json.loads(line)
+
+
+def get_traits_from_json(clinvar_json, trait_dict):
+
+    # This if-else block is due to the change in the format of the CellBase JSON that holds the
+    # ClinVar data. Originally "clinvarSet" was the top level, but this level was removed and
+    # referenceClinVarAssertion is now the top level.
+    if "clinvarSet" in clinvar_json:
+        trait_set = clinvar_json["clinvarSet"]["referenceClinVarAssertion"]["traitSet"]
+    else:
+        trait_set = clinvar_json["referenceClinVarAssertion"]["traitSet"]
```

Comment:
What about extracting this if-else to another function (nested call) to make the core process more straightforward?